torquedata/torquedata/routes.py

```diff
@@ -109,20 +111,44 @@ def sheet_toc(sheet_name, toc_name, fmt):
     else:
         raise Exception("Only mwiki format valid for toc")
 
-@app.route('/api/<sheet_name>/id/<key>.<fmt>')
-def row(sheet_name, key, fmt):
-    group = request.args.get("group")
-    wiki_key = request.args.get("wiki_key")
-
-    valid_id = False
 
+def is_valid_id(group, wiki_key, key):
     if wiki_key in permissions[sheet_name].keys() and group in permissions[sheet_name][wiki_key].keys():
         if "valid_ids" not in permissions[sheet_name][wiki_key][group].keys():
-            valid_id = True
+            return True
         else:
-            valid_id = (key in permissions[sheet_name][wiki_key][group]["valid_ids"])
+            return (key in permissions[sheet_name][wiki_key][group]["valid_ids"])
 
-    if not valid_id:
+
+@app.route('/api/<sheet_name>/edit-record/<key>', methods=['POST'])
+def edit_record(sheet_name, key):
+    group = request.json.get("group")
+    wiki_key = request.json.get("wiki_key")
+    new_values = json.loads(request.json.get("new_values"))
+
+    if not is_valid_id(group, wiki_key, key):
+        abort(403)
+
+    for field, val in new_values.items():
+        edit = {
+            "new_value": val,
+            "edit_message": "",
```

Comment:
If there's no edit message because the UI option to provide one doesn't exist, this should probably be `None` (i.e., NULL) rather than `""`.  Later, when the UI option is present but someone offers an empty edit message, then that's when we use `""`.